src/test/versionbits_tests.cpp

```diff
@@ -14,6 +14,15 @@
 /* Define a virtual block time, one block per 10 minutes after Nov 14 2014, 0:55:36am */
 static int32_t TestTime(int nHeight) { return 1415926536 + 600 * nHeight; }
 
+static constexpr const char* StateName(ThresholdState state) {
+    return (state == ThresholdState::DEFINED ?     "DEFINED" :
+            state == ThresholdState::STARTED ?     "STARTED" :
+            state == ThresholdState::LOCKED_IN ?   "LOCKED_IN" :
+            state == ThresholdState::ACTIVE ?      "ACTIVE" :
+            state == ThresholdState::FAILED ?      "FAILED" :
+            "");
```

Comment:
Any reason to disable the compile time checks that come with a switch-case?

Reply:
IIUC it's to remain compatible with C++11: https://github.com/bitcoin/bitcoin/pull/19573/commits/bae9a452191a7a83478f7d508a54f4a04d385505#r579334677

Reply:
I am pretty sure that `switch` exists way before C++11

Reply:
It can't be a constexpr function with a switch. I'm not sure what the reason for it being constexpr was, though; I may have been trying something with templates at one point?

Reply:
I can change it to use switch and not constexpr. Is there a `./configure` flag to force c++11?

Reply:
@Sjors Automatic if you develop on top of 831675c8dccfa6525ffe751da3cc60709c380953 ;)

Reply:
```c++
static const std::string StateName(ThresholdState state)
{
    switch (state) {
    case ThresholdState::DEFINED:   return "DEFINED";
    case ThresholdState::STARTED:   return "STARTED";
    case ThresholdState::LOCKED_IN: return "LOCKED_IN";
    case ThresholdState::ACTIVE:    return "ACTIVE";
    case ThresholdState::FAILED:    return "FAILED";
    } // no default case, so the compiler can warn about missing cases
    return "";
}
```

looks fine to me fwiw.

Reply:
Two birds, one stone...